src/main/java/hina/HinaBot.java

```diff
@@ -0,0 +1,118 @@
+package hina;
+
+import hina.exceptions.HinaException;
+import hina.task.Deadline;
+import hina.task.Event;
+import hina.task.Task;
+
+import java.util.Scanner;
+
+public class HinaBot {
+    protected static Task[] taskList = new Task[100];
+    protected static int taskCount = 0;
+
+    public static void main(String[] args) {
+        String line;
+        Scanner in = new Scanner(System.in);
+        showGreeting();
+        while (true) {
+            try {
+                line = in.nextLine();
+                handleCommand(line);
+            } catch (HinaException cmdException) {
+                System.out.println(">.< Hina does not recognise this command!");
+            } catch (StringIndexOutOfBoundsException argException) {
+                System.out.println("@_@ Please give Hina more details!");
+            }
+        }
+    }
+
+    private static void showGreeting() {
+        System.out.println("Hello master!");
+        System.out.println("What are your orders?");
+    }
+
+    public static void handleCommand(String command) throws HinaException {
+        if (command.equalsIgnoreCase("bye")) {
+            System.out.println("Goodbye master, let's meet again soon...");
+            System.exit(0);
+        } else if (command.equalsIgnoreCase("list")) {
+            listTasks();
+        } else if (command.split(" ")[0].equalsIgnoreCase("mark")) {
+            int taskIndex = Integer.parseInt(command.split(" ")[1]);
+            markTask(taskIndex);
+        } else if (command.split(" ")[0].equalsIgnoreCase("unmark")) {
+            int taskIndex = Integer.parseInt(command.split(" ")[1]);
+            unmarkTask(taskIndex);
+        } else if (command.split(" ")[0].equalsIgnoreCase("todo")) {
+            addTask(command.substring(5));
+        } else if (command.split(" ")[0].equalsIgnoreCase("deadline")) {
+            addDeadline(command.substring(9));
+        } else if (command.split(" ")[0].equalsIgnoreCase("event")) {
+            addEvent(command.substring(6));
+        } else {
+            throw new HinaException();
+        }
+    }
```

Comment:
Good job of splitting a big method into smaller one and improving your code quality